### ISSUE TITLE
[expo-camera] Don't receive cornerPoints / bounds on Android

### ISSUE THREAD

**Issue by gerritwalther** [expo/expo#17795]
### Summary

In the documentation for expo-camera for Expo SDK 45 it says that the [onBarCodeScanned](https://docs.expo.dev/versions/v45.0.0/sdk/camera/#onbarcodescanned) callback now returns a BarCodeScanningResult. This should (in most cases as I understand) also contain the cornerPoints (and bounds). In the linked snack below I won't get those cornerPoints using a Samsung A51.
Tried it on an iPhone and that returns the cornerPoints and bounds as expected.

I also checked the BarCodeScanner module with my Android phone and I do get cornerPoints and bounds.

I've also had a look at the source code, but I'm not that familiar with Kotlin to find the problem.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

45

### Environment

  expo-env-info 1.0.3 environment info:
    System:
      OS: Linux 5.10 Manjaro Linux
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.2.0 - /usr/bin/node
      Yarn: 1.22.19 - /usr/bin/yarn
      npm: 8.5.5 - /usr/bin/npm
    SDKs:
      Android SDK:
        API Levels: 29, 30, 31
        Build Tools: 29.0.2, 29.0.3, 30.0.0
        System Images: android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: AI-212.5712.43.2112.8609683
    npmPackages:
      expo: ~45.0.0 => 45.0.5 
      react: 17.0.2 => 17.0.2 
      react-dom: 17.0.2 => 17.0.2 
      react-native: 0.68.2 => 0.68.2 
      react-native-web: 0.17.7 => 0.17.7 
    npmGlobalPackages:
      expo-cli: 5.4.8
    Expo Workflow: managed

### Reproducible demo

https://snack.expo.dev/@gerrit_w/no-cornerpoints-on-android

**Comment by Simek** [comment]
Hello @gerritwalther, `cornerPoints` are marked as an optional field, which means that not on every device those fields will be populated and we do not guarantee the availability of those data.

<img width="517" alt="Screenshot 2022-06-13 at 10 46 17" src="https://user-images.githubusercontent.com/719641/173315607-18ee9479-76f6-4417-9fcb-d6a0dbe4a2de.png">

This usually is due to hardware or OS APIs limitation on the device, and we cannot do much about that.

**Comment by gerritwalther** [comment]
Thank you for your answer @Simek . But if the BarCodeScanner module can return those cornerPoints, the Camera module should return the cornerPoints as well?

**Comment by Simek** [comment]
> if the BarCodeScanner module can return those cornerPoints, the Camera module should return the cornerPoints as well?

@gerritwalther I'm not sure about that, those packages probably use slightly different native APIs underneath, so the results might vary, but I will investigate this soon. 

I also did not understand correctly the problem you have, let me rephrase it, and please confirm if I'm right:
* when using `BarCodeScanner` component from `expo-barcode-scanner` corner point data are available, but when you use the `Camera` component from `expo-camera` instead those data are missing for the same hardware device.

Also listing the device which you have tested and work/do not work could be helpful.

**Comment by gerritwalther** [comment]
>     * when using `BarCodeScanner` component from `expo-barcode-scanner` corner point data are available, but when you use the `Camera` component from `expo-camera` instead, those data are missing for the same hardware device, right?

Yes, correct.

> If the description of the problem above is correct, please update the included Snack, so it reflects the situation (add both components, allow to switch them and compare data).

I've updated the Snack and you can now switch between `BarCodeScanner` and `Camera` components. 

> Also listing the device which you have tested and work/do not work could be helpful.

The device I was using is a Samsung A51.

**Comment by gerritwalther** [comment]
One more thing: It makes no difference whether I add `useCamera2Api={true}` or leave it on the default with `false`.

**Comment by MateuszJaniszewskiSof** [comment]
Same here (SDK 46). Can' receive `cornerPoints` on:

- BlackView 6300Pro - Android 10 
- Galaxy A71 - Android 12

On both devices, it's working fine when using `BarCodeScanner` but I can't use it because of a lack of `torch` support.

**Comment by igoro00** [comment]
Same here on Xiaomi Mi6 (Android 9), `BarCodeScanner` works, `Camera` doesn't(but it has other features I wanna use that don't exist in `BarCodeScanner`). I'll test other phones(Android and iOS) later today

**Comment by igoro00** [comment]
I went into [`packages/expo-barcode-scanner/android/src/main/.../BarcodeScannerProvider.kt` ](https://github.com/expo/expo/blob/main/packages/expo-barcode-scanner/android/src/main/java/expo/modules/barcodescanner/BarCodeScannerProvider.kt)and removed GMV scanner as an option forcing expo-barcode-scanner to use zxing. It still scanned fine but stopped returning cornerPoints(just like expo-camera!).
I couldn't find a file where expo-camera calls `GMVBarCodeScanner` or `ZxingBarCodeScanner` yet but I think expo-camera might use zxing scanner exclusively.

Also, in may/june 2021 google deprecated Google Mobile Vision (GMV) for ML Kit. Are there any plans to move expo-camera to ML Kit?

**Comment by deywersonp** [comment]
Heyo! 
The problem above persists on `SDK 46`.

I'm using Expo SDK 46 and I'm not receiving `cornerPoints/bounds` when using `Camera` from `expo-camera`, but I receive this information when using `BarCodeScanner` from `expo-barcode-scanner`.

**Comment by deywersonp** [comment]
> > Heyo!
> > The problem above persists on `SDK 46`.
> > I'm using Expo SDK 46 and I'm not receiving `cornerPoints/bounds` when using `Camera` from `expo-camera`, but I receive this information when using `BarCodeScanner` from `expo-barcode-scanner`.
> 
> This pull request was submitted and merged way after SDK 46 was released. Wait for sdk 47 like me or find a way to use expo-camera v13.0.0 with sdk 46

Oh, I didn't know. Thanks for the clarification!

**Comment by kaden-sharpin** [comment]
SDK 47 is now in beta. If you want to try it, run `EXPO_BETA=1 expo-cli upgrade`. 

source: https://blog.expo.dev/expo-sdk-47-beta-is-now-available-fa760eea1908

**Comment by gerritwalther** [comment]
I can confirm that my Samsung A51 now is getting the boundingBox / cornerPoints on each scan in the Camera module.

Thank you for fixing the issue @igoro00
